test/instrumentation/modules/@elastic/elasticsearch.test.js

```diff
@@ -686,8 +673,8 @@ function checkSpanOutcomesFailures (t) {
   return function (data) {
     data.spans.sort((a, b) => { return a.timestamp < b.timestamp ? -1 : 1 })
     if (semver.gte(esVersion, '7.14.0') && semver.satisfies(esVersion, '7.x')) {
-      // Remove leading ES span and HTTP span from product check.
-      data.spans = data.spans.slice(2)
+      // Remove the product check spans for subsequent assertions.
+      data.spans = data.spans.slice(0, 1).concat(data.spans.slice(3))
```

Comment:
REVIEW NOTE: The order of product-check spans has changed due to the change in the implementation of the instrumentation. We create a Span on `Transport#request()` now, rather than later when the "request" diagnostic event is sent. That means that the first Span is the one we want to keep.

1. ES span for the actual client command (want to keep this one)
2. ES span for the product check
3. HTTP span for the product check
4. HTTP span for the actual client command  (want to keep this one)